pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,237 @@
+    """
+    if isinstance(wires, int):
+        wires = [wires]
+
+    # We compute the `coeffs` and `observables` of the laser field
+    coeffs = [rabi, detuning]
+    rabi_observable = sum(
+        qml.math.cos(phase) * qml.PauliX(wire) - qml.math.sin(phase) * qml.PauliY(wire)
+        for wire in wires
+    )
+    detuning_observable = sum(qml.PauliZ(wire) for wire in wires)
+    observables = [rabi_observable, detuning_observable]
+
+    # We convert the pulse data into a list of ``RydbergPulse`` objects
+    pulses = [RydbergPulse(rabi, phase, detuning, wires)]
+    return RydbergHamiltonian(coeffs, observables, pulses=pulses)
+
+
+class RydbergHamiltonian(ParametrizedHamiltonian):
+    r"""Class representing the Hamiltonian of an ensemble of Rydberg atoms under the action of
+    local and global laser fields:
+
+    .. math::
+
+        H = \hbar \frac{1}{2} \sum_i  \Omega_i(t) (\cos(\phi)\sigma_i^x - \sin(\phi)\sigma_i^y) -
+        \frac{1}{2} \sum_i \delta_i(t) \sigma_i^z + \sum_{i<j} V_{ij} n_i n_j
+
+    where :math:`\Omega_i` and :math:`\delta_i` correspond to the amplitude and detuning of the
+    laser applied to atom :math:`i`, and :math:`\sigma^\alpha` for :math:`\alpha = x,y,z` are
+    the Pauli matrices.
+
+    The last term of the sum corresponds to the interaction between the atoms of the ensemble
+    due to the Rydberg blockade. For more info, check :func:`rydberg_interaction`.
+
+    .. note::
+
+        :math:`\hbar` is set to 1.
+
+    .. warning::
+
+        When adding a ``RydbergHamiltonian`` with a :class:`ParametrizedHamiltonian` all the
+        information needed to translate this class into hardware will be lost.
+
+    .. seealso:: :class:`rydberg_interaction`, :class:`rydberg_transition`
+
+
+    Args:
+        coeffs (Union[float, callable]): coefficients of the Hamiltonian expression, which may be
+            constants or parametrized functions. All functions passed as ``coeffs`` must have two
+            arguments, the first one being the trainable parameters and the second one being time.
+        observables (Iterable[Observable]): observables in the Hamiltonian expression, of same
+            length as ``coeffs``
+
+    Keyword Args:
+        register (list): list of coordinates (in micrometers) of each atom in the ensemble
+        transition (list): information about the amplitude, phase, detuning and wires of the pulses
+        interaction_coeff (float): Rydberg interaction constant in units: :math:`MHz \times \mu m^6`.
+            Defaults to :math:`862690 \times 2\pi MHz \times \mu m^6`.
+
+    Returns:
+        RydbergHamiltonian: class representing the Hamiltonian of an ensemble of Rydberg atoms
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        coeffs,
+        observables,
+        register: list = None,
+        pulses: List["RydbergPulse"] = None,
```

Comment:
```suggestion
        pulses: List[RydbergPulse] = None,
```

Reply:
You can't do this because the class `RydbergPulse` is defined after the `RydbergHamiltonian` class.

Python lets you use strings to specify types that are not yet defined. If you don't like this we can always move the `RydbergPulse` class to the beginning of the file.

Reply:
Ah thanks I didn't realize.